specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+
+# Full API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass AppNotificationTextProperties
+    {
+        // Contains the set of <text> attributes
+        AppNotificationTextProperties();
+
+        AppNotificationTextProperties SetLanguage(String value);
+        AppNotificationTextProperties UsesCallScenarioAlign();
+        AppNotificationTextProperties SetMaxLines(Int32 value);
+
+        String GetXml();
+    }
+
+    enum ButtonColor
+    {
+        Red, // The button will be green
+        Green, // The button will be red
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
+        // to add arguments to the button.
+        AppNotificationButton();
+        AppNotificationButton(String buttonText);
+
+        AppNotificationButton AddArgument(String key, String value);
+
+        // Sets the IconUri for the button.
+        AppNotificationButton SetIcon(Windows.Foundation.Uri value);
+
+        // The tooltip for a button, if the button has an empty content string.
+        AppNotificationButton SetToolTip(String value);
+
+        // Sets the Button as context menu action.
+        AppNotificationButton UseContextMenuPlacement();
+
+        // Sets the ButtonColor to Green or Red
+        AppNotificationButton SetButtonColor(ButtonColor value);
+
+        // Sets the ID of an existing TextBox in order to have this button display to the right of the input.
+        AppNotificationButton SetInputId(String value);
+
+        // Launches the URI passed into the button when activated.
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri);
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri, String targetAppId);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationComboBox
+    {
+        AppNotificationComboBox(String id);
+
+        // Add a selection item to the AppNotificationComboBox if InputType is AppNotificationComboBox. The id parameter is used to retrieve the user input when the app is activated
+        // and content is the displayed text for the item.
+        AppNotificationComboBox AddItem(String id, String content);
+
+        // Adds a title to display on top
+        AppNotificationComboBox SetTitle(String value);
+
+        // Sets the default selection to be displayed by the AppNotificationComboBox
+        AppNotificationComboBox SetSelectedItem(String id);
+
+        // Retrieves the XML content of the input.
+        String GetXml();
```

Comment:
same comment here and elsewhere about GetXml being string versus GetXmlText

Reply:
GetXml going to be removed and replaced by BuildNotification.